R/check_freq_method.R

```diff
@@ -0,0 +1,15 @@
+check_freq_method <- function(wsmaf, plmaf, seq_error) {
+  # Compute number of variant sites
+  variant <- ifelse(wsmaf <= seq_error | wsmaf >= (1 - seq_error), 0, 1)
+  n_variant <- sum(variant)
+
+  # Compute expected number of variant sites using Hardy-Weinberg
+  hardy_weinberg <- 2 * plmaf * (1 - plmaf)
+  n_loci <- length(plmaf)
+
+  bin_ci <- Hmisc::binconf(sum(hardy_weinberg), n_loci) * n_loci
+
+  # If the actual number of variant sites is less than the lower bound of the CI
+  # return FALSE, otherwise return FALSE
+  if (n_variant < bin_ci[2]) FALSE else TRUE
```

Comment:
FYI: doing something like `bin_ci[1, "Lower"]` will make it clearer that it is the lower bound being taken and makes it slightly more future error proof if the order changes. (Suppose equally the order may stay the same but the column names would change)